src/core/unit.c

```diff
@@ -3633,16 +3635,50 @@ int unit_install_bus_match(Unit *u, sd_bus *bus, const char *name) {
         if (r < 0)
                 return r;
 
-        r = sd_bus_call_method_async(
-                        bus,
-                        &u->get_name_owner_slot,
-                        "org.freedesktop.DBus",
-                        "/org/freedesktop/DBus",
-                        "org.freedesktop.DBus",
-                        "GetNameOwner",
-                        get_name_owner_handler,
-                        u,
-                        "s", name);
+        r = sd_bus_message_new_method_call(bus,
+                                           &m1,
+                                           "org.freedesktop.DBus",
+                                           "/org/freedesktop/DBus",
+                                           "org.freedesktop.DBus",
+                                           "AddMatch");
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_append(m1, "s", match);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_call_async(bus,
+                              &u->match_bus_slot,
+                              m1,
+                              signal_name_owner_changed,
+                              u,
+                              DEFAULT_TIMEOUT_USEC);
```

Comment:
hmm, i think it would be correct to use the start timeout of the unit here. i.e. the timeout_start_usec field of the Service object. after all we are doing this for putting a time limit on the logic that detects when a service finished starting up.

Now, this here is the generic Unit code, not the specific Service code that maintains the timeout. Hence, maybe simply pass the right value down as a function parameter to unit_install_bus_match() and so on

Reply:
Also, please add a comment here explaining this